### PR TITLE
Basic services table implemented

### PR DESCRIPTION
**Pull request by philrenaud**
- Moves existing `service` model to `service-fragment` model
  - ditto serializer
- Adds a rudimentary table on the /services route
- Normalizes services API endpoint response to be a flattened array

image

### CODE REVIEW

**Review by ChaiWithJai — CHANGES_REQUESTED** [review]
Hey, buddy strong start! Some blocking feedback from me:

1. Can we have this PR merge into a trunk branch instead of main?
2. Can we handle the creation of the `Service` model and `ServiceFragment` refactoring into its own PR? I support atomic PRs, but in this case, I'd suggest breaking this PR down as follows:

- `ServiceFragment` model refactoring with regression testing. (isolate refactoring)
- `Service` model creation to what we have in the backend. And associated `Adapter` and `Serializer` logic. (isolate core logic)
- Handle basic `ServicesTable` implementation. (view layer implementation)

This way you can make sure that we're not adding `namespaces` later. We need to implement them along the entire process.

File: ui/app/models/service.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-import Fragment from 'ember-data-model-fragments/fragment';
-import { fragment } from 'ember-data-model-fragments/attributes';
+// import { fragment } from 'ember-data-model-fragments/attributes';
+import Model from '@ember-data/model';
+import { alias } from '@ember/object/computed';
 
-export default class Service extends Fragment {
-  @attr('string') name;
-  @attr('string') portLabel;
+export default class Service extends Model {
```
[the remark this answers is not in the document]
Comment (philrenaud) [reply ui/app/models/service.js:6]: Definitely agreed, but as the scope of this is limited to just the main /services page, and it only returns ServiceName and Tags, that's all I'm extending to the model here.

As we build out more of the UI that fetches and gets more properties, those'll be added to the model.

[the remark this answers is not in the document]
Comment (philrenaud) [reply ui/app/models/service.js:6]:
```
[{"Namespace":"default","Services":[{"ServiceName":"example","Tags":["foo","bar"]},{"ServiceName":"example2","Tags":["thelonious","max"]}]}]
```

[the remark this answers is not in the document]
Comment (philrenaud) [reply ui/app/models/service.js:6]: Answered this elsewhere, but these are not properties that return in the list view for Services. As I build out pieces of the UI that use things like nodeID, alloId, and job, I'll model those as well.

File: ui/app/models/service-fragment.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import Fragment from 'ember-data-model-fragments/fragment';
+import { fragment } from 'ember-data-model-fragments/attributes';
+
+export default class Service extends Fragment {
+  @attr('string') name;
+  @attr('string') portLabel;
+  @attr() tags;
+  @attr('string') onUpdate;
+  @fragment('consul-connect') connect;
+}
```
Comment (ChaiWithJai) [inline ui/app/models/service-fragment.js:11]: This is consumed by `TaskGroup` and in the associated `Allocation`, `TaskGroup` and `Task` pages. I'm noticing that CI detected 19 failures and error messages failing on tests like:

`allocation detail: if stopping or restarting fails, an error message is shown`.

Any idea why these tests are failing CI?

File: ui/app/models/service.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChaiWithJai) [reply ui/app/models/service.js:6]: Generally, I'm not a fan of modeling as you go. And I think not modeling `Variables` specifically for building the URL ended up making that work downstream much more difficult. I'm already noticing a decent number of lurking bugs that can arise later if we don't take the time to model correctly at first.

**Review by ChaiWithJai — APPROVED** [review]

File: ui/app/models/service-fragment.js (hunk 2 above)
Comment (philrenaud) [reply ui/app/models/service-fragment.js]: Yep! Had forgotten to rename the mirage model: https://github.com/hashicorp/nomad/pull/13856/commits/3cf40b711a3615384765e47e9fa847aa9d4755f2

now settled

File: ui/app/templates/services/index.hbs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+				<td>{{row.model.name}}</td>
+				<td>
+					{{#each row.model.tags as |tag|}}
+						<span class="tag">{{tag}}</span>
+					{{/each}}
+				</td>
+			</tr>
+		</t.body>
+	</ListTable>
+</section>
```
Comment (ChaiWithJai) [inline ui/app/templates/services/index.hbs:26]: `question`: Is this a duplicate template? I see we have nearly the same template at `templates/services/index`? Why are we adding a duplicate of this to `templates/components/services.hbs`?

**Review by ChaiWithJai — COMMENTED** [review]
Noticing a few places where we might be able to trim this PR. Found a duplicate template. And also noticed that the `service` relationship in the `job` model is unused unless I missed something.

**Comment by philrenaud** [comment]
Closing; handled in a very different way in https://github.com/hashicorp/nomad/pull/14408

**Pull request closed without merging**